client/src/utils/CSSHelp.ts

```diff
@@ -0,0 +1,144 @@
+const CSSTypes = {
+  style: 1,
+  import: 3,
+  media: 4,
+  fontface: 5,
+  keyframes: 7
+};
+
+type Rule<T> = T[] | undefined;
+
+export interface ExtendedStyleRule extends CSSStyleRule {
+  isDeclaredAfter: (selector: string) => boolean;
+}
+
+// TODO: add position property to reduce logic in isDeclaredAfter
+Object.defineProperty(CSSStyleRule.prototype, 'position', {
+  set: function () {
+    const cssRules = this.parentStyleSheet?.cssRules as
+      | CSSStyleRule
+      | undefined;
+    const cssRulesArr = Array.from(cssRules);
```

Comment:
```suggestion
    const cssRules = this.parentStyleSheet?.cssRules as
      | CSSStyleRule
      | undefined;
    const cssRulesArr = Array.from(cssRules);
```

Issue here is that `| undefined` typing.

Can't pass `undefined` into `Array.from` which is why TS is mad - we should use an empty array fallback probably.

Reply:
I did realise that, but adding the empty `|| []` just yields another error.

Reply:
Ah, then we run in to the issue of `CSSStyleRule` missing the `length` property. 🤔

Reply:
As I said in chat, this is really heavily exposing the implementation details.  We really shouldn't need to even know that `CSSStyleRule`s exist in order to use these tools.

 Also, there's no need to modify the prototype here.  We can just define our own data structures.